### PR TITLE
Coupon view tests update.

### PR DESCRIPTION
**Pull request by vkaracic**

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @vkaracic! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

<a   href="https://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fecommerce&number=1212"   rel="nofollow" >Create an OSPR issue for this pull request</a>.

File: ecommerce/extensions/voucher/tests/test_utils.py [hunk 1]
```diff
@@ -568,3 +572,29 @@ def test_update_voucher_offer(self):
...
+        self.assertEqual(voucher.code, VOUCHER_CODE)
+        self.assertEqual(len(products), 1)
+        self.assertEqual(products[0], original_product)
+
+    def test_no_product(self):
+        """ Verify that an exception is raised if there is no product. """
+        code = FuzzyText().fuzz()
+        voucher = VoucherFactory(code=code)
+        offer = ConditionalOfferFactory()
+        voucher.offers.add(offer)
```
Comment (marjev) [inline ecommerce/extensions/voucher/tests/test_utils.py:592]: Can you use https://github.com/edx/ecommerce/blob/master/ecommerce/extensions/test/factories.py#L54 instead?

Comment (vkaracic) [reply ecommerce/extensions/voucher/tests/test_utils.py:592]: No, that method creates a product and adds it to the voucher range, the test ensures that an exception is raised if there is no product.

**Review by marjev — APPROVED** [review]

File: ecommerce/coupons/tests/test_views.py [hunk 2]
```diff
@@ -47,75 +44,50 @@
...
 
-    def setUp(self):
-        super(CouponAppViewTests, self).setUp()
-
     def test_login_required(self):
         """ Users are required to login before accessing the view. """
         self.client.logout()
         response = self.client.get(self.path)
-        self.assertEqual(response.status_code, 302)
+        self.assertEqual(response.status_code, status.HTTP_302_FOUND)
```
Comment (clintonb) [inline ecommerce/coupons/tests/test_views.py]: Considering this isn't an API module, there shouldn't be any references to DRF here. Even if this were an API module, I see no reason to introduce an unnecessary dependency in our tests.

File: ecommerce/coupons/tests/test_views.py [hunk 3]
```diff
@@ -47,75 +44,50 @@
...
-            get_voucher_and_products_from_code(code=code)
 
-    def test_get_non_existing_voucher(self):
-        """ Verify that get_voucher_and_products_from_code() raises exception for a non-existing voucher. """
-        with self.assertRaises(Voucher.DoesNotExist):
-            get_voucher_and_products_from_code(code=FuzzyText().fuzz())
+class VoucherIsValidTests(CourseCatalogTestMixin, TestCase):
+    def setUp(self):
+        super(VoucherIsValidTests, self).setUp()
+        self.request = RequestFactory().request()
```
Comment (clintonb) [inline ecommerce/coupons/tests/test_views.py]: The base `TestCase` class already includes `self.request`.

File: ecommerce/coupons/tests/test_views.py [hunk 4]
```diff
@@ -124,57 +96,54 @@ def test_future_voucher(self):
...
-        voucher, product = prepare_voucher(code=COUPON_CODE, usage=Voucher.ONCE_PER_CUSTOMER, max_usage=1)
+        voucher, product = prepare_voucher(usage=Voucher.ONCE_PER_CUSTOMER, max_usage=1)
         user = self.create_user()
         error_msg = 'This coupon code is no longer available.'
         self.assert_error_messages(voucher, product, user, error_msg)
 
     def test_used_voucher(self):
-        """ Verify voucher_is_valid() assess that the voucher is unavailable. """
-        voucher, product = prepare_voucher(code=COUPON_CODE)
+        """Used voucher should not be available."""
```
Comment (clintonb) [inline ecommerce/coupons/tests/test_views.py:146]: Verify used vouchers are not valid.

File: ecommerce/coupons/tests/test_views.py [hunk 5]
```diff
@@ -215,15 +169,16 @@ def test_invalid_voucher(self):
...
     def test_expired_voucher(self):
         """ Verify proper response is returned for expired vouchers. """
+        code = FuzzyText().fuzz()
         start_datetime = now() - datetime.timedelta(days=20)
         end_datetime = now() - datetime.timedelta(days=10)
-        prepare_voucher(code='EXPIRED', start_datetime=start_datetime, end_datetime=end_datetime)
+        prepare_voucher(code=code, start_datetime=start_datetime, end_datetime=end_datetime)
 
-        url = self.path + '?code={}'.format('EXPIRED')
+        url = self.path + '?code={}'.format(code)
```
Comment (clintonb) [inline ecommerce/coupons/tests/test_views.py]: Use `str.format()` for the entire string.

File: ecommerce/coupons/tests/test_views.py [hunk 6]
```diff
@@ -249,33 +204,32 @@ def prepare_url_for_credit_seat(self, code='CREDIT', enterprise_customer=None):
         # if one of them is missing it could result in a SEGFAULT error when running tests
         # with migrations enabled.
         range_kwargs = {
-            'products': [credit_seat, ],
+            'products': [credit_seat],
             'course_seat_types': 'credit',
             'catalog_query': '*:*',
+            'enterprise_customer': enterprise_customer
```
Comment (clintonb) [inline ecommerce/coupons/tests/test_views.py]: Add a trailing comma.

File: ecommerce/coupons/tests/test_views.py [hunk 7]
```diff
@@ -249,33 +204,32 @@ def prepare_url_for_credit_seat(self, code='CREDIT', enterprise_customer=None):
...
-            range_kwargs['enterprise_customer'] = enterprise_customer
         _range = RangeFactory(**range_kwargs)
         prepare_voucher(code=code, _range=_range)
 
         url = '{path}?code={code}'.format(path=self.path, code=code)
         return url
 
     def test_redirect_to_login(self):
-        """ Verify anonymous user gets redirected to login page. """
+        """User needs be logged in to open offer page with credit seats."""
```
Comment (clintonb) [inline ecommerce/coupons/tests/test_views.py]: Verify users need to be logged in to view the offer page.

File: ecommerce/coupons/tests/test_views.py [hunk 8]
```diff
@@ -249,33 +204,32 @@ def prepare_url_for_credit_seat(self, code='CREDIT', enterprise_customer=None):
...
         self.client.logout()
         url = self.prepare_url_for_credit_seat()
         response = self.client.get(url)
 
         testserver_login_url = self.get_full_url(reverse('login'))
         expected_url = '{path}?next={next}'.format(path=testserver_login_url, next=urllib.quote(url))
-        self.assertRedirects(response, expected_url, target_status_code=302)
+        self.assertRedirects(response, expected_url, target_status_code=status.HTTP_302_FOUND)
```
Comment (clintonb) [inline ecommerce/coupons/tests/test_views.py]: See comments above about unnecessary DRF dependency.

File: ecommerce/coupons/views.py [hunk 9]
```diff
@@ -56,7 +56,7 @@ def voucher_is_valid(voucher, products, request):
     """
 
     if voucher is None:
-        return False, _('Coupon does not exist')
+        return False, _('Coupon does not exist.')
```
Comment (clintonb) [inline ecommerce/coupons/views.py:60]: Update translations.

**Review by clintonb — CHANGES_REQUESTED** [review]

File: ecommerce/coupons/tests/test_views.py (hunk 4 above)
Comment (vkaracic) [reply ecommerce/coupons/tests/test_views.py:146]: The "verify" part is redundant in test docstrings, and I'd like to follow what these guys are doing: https://code.djangoproject.com/ticket/27392

**Review by clintonb — APPROVED** [review]

**Pull request merged**